storage-operator/pkg/controller/volume/volume_controller.go

```diff
@@ -196,24 +725,91 @@ func nodeAffinity(node types.NodeName) *corev1.VolumeNodeAffinity {
 		Required: &selector,
 	}
 	return &affinity
+}
+
+// Destroy the give PersistentVolume.
+func (self *ReconcileVolume) destroyPersistentVolume(
+	ctx context.Context,
+	request reconcile.Request,
+	volume *storagev1alpha1.Volume,
+	pv *corev1.PersistentVolume,
+) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Name", request.Name)
+	nodeName := string(volume.Spec.NodeName)
+
+	switch volume.Status.Job {
+	case "": // No job in progress: call Salt to unprepare the volume.
+		if jid, err := self.salt.UnprepareVolume(ctx, nodeName); err != nil {
+			reqLogger.Error(err, "failed to run UnprepareVolume")
+			return delayedRequeue(err)
+		} else {
+			reqLogger.Info("start to unprepare the volume")
+			self.recorder.Event(
+				volume, corev1.EventTypeNormal, "SaltCall",
+				"Salt call to UnprepareVolume started",
+			)
+			return self.setTerminatingVolumeStatus(ctx, volume, jid)
+		}
+	case JOB_DONE_MARKER: // Salt job is done, now let's delete the PV.
+		if err := self.client.Delete(ctx, pv); err != nil {
+			reqLogger.Error(
+				err, "cannot delete PersistentVolume: requeue",
+				"PersistentVolume.Name", volume.Name,
+			)
+			return delayedRequeue(err)
+		}
+		reqLogger.Info(
+			"deleting backing PersistentVolume",
+			"PersistentVolume.Name", pv.Name,
+		)
+		if err := self.removePvFinalizer(ctx, pv); err != nil {
+			reqLogger.Error(err, "cannot remove PersistentVolume finalizer")
+			return delayedRequeue(err)
+		}
+		reqLogger.Info("PersistentVolume finalizer removed")
+		self.recorder.Event(
+			volume, corev1.EventTypeNormal, "PvDeletion",
+			"backing PersistentVolume deleted",
+		)
+		return requeue(nil)
+	default: // UnprepareVolume in progress: poll its state.
+		return self.pollSaltJob(
+			ctx, "UnprepareVolume", volume, self.setTerminatingVolumeStatus,
+		)
+	}
+}
 
+// Trigger a reschedule after a short delay.
+func delayedRequeue(err error) (reconcile.Result, error) {
+	delay := 1 * time.Second
```

Comment:
I think this delay is too low. I'd consider a 10s timeout, and over time something with a  backoff.

Consider a 6 node system with 100 disks per node, where we create all Volumes at once (e.g. at initial deployment time). We'd be hammering the whole system quite a bit!

Reply:
Done (10s for now, we can implement backoff later).